api/src/plugins/redirect-with-message.ts

```diff
@@ -0,0 +1,54 @@
+import { FastifyPluginCallback, type FastifyReply } from 'fastify';
+import fp from 'fastify-plugin';
+// TODO: (POST MVP)use node's querystring and just JSON stringify the message.
+// No need for query-string on either side.
+
+import qs from 'query-string';
+
+declare module 'fastify' {
+  interface FastifyReply {
+    redirectWithMessage: typeof redirectWithMessage;
+  }
+}
+
+type Message = {
+  type: 'info' | 'danger' | 'success' | 'errors';
+  content: string;
+};
+
+type MessageQuery = {
+  info?: string[];
+  danger?: string[];
+  success?: string[];
+  errors?: string[];
+};
+
+// The client expects a message like { info: ['foo'] }, { danger: ['bar'] } etc.
+const prepareMessage = (message: Message): MessageQuery => ({
+  [message.type]: [message.content]
+});
```

Comment:
All of this so we can add content as `string` and not `array` 🤣.

This seems like something that should be simpler, is it okay to create an issue for this and we tackle it post mvp?

Reply:
The whole thing should be simpler, yeah, but I wasn't sure if we were going to phase this out after the MVP.  So, I left a few comments and we can discuss further depending on the requirements.